warehouse/query-core/src/main/java/org/apache/commons/jexl2/DatawaveJexlScript.java

```diff
@@ -0,0 +1,171 @@
+package org.apache.commons.jexl2;
+
+import datawave.query.jexl.visitors.TreeFlatteningRebuildingVisitor;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+/**
+ * Lifted from {@link ExpressionImpl}. Modified to flatten the resulting {@link ASTJexlScript}.
+ */
+public class DatawaveJexlScript implements Expression, Script {
```

Comment:
I see a lot of the methods are cut and paste from the Jexl implementations.  Can we extend instead or implement using aggregation instead?  Just wondering.

Reply:
I tried extending/implementing and didn't find that to be particularly successful. I could take another look, this was written pretty quickly.